dspace-spring-rest/src/main/java/org/dspace/app/rest/security/StatelessLoginFilter.java

```diff
@@ -46,12 +61,29 @@ public Authentication attemptAuthentication(HttpServletRequest req,
         String user = req.getParameter("user");
         String password = req.getParameter("password");
 
-        return authenticationManager.authenticate(
-                new DSpaceAuthentication(
-                        user,
-                        password,
-                        new ArrayList<>())
-        );
+        try {
+             return authenticationManager.authenticate(
+                    new DSpaceAuthentication(
+                            user,
+                            password,
+                            new ArrayList<>()
+                    )
+            );
+        } catch(BadCredentialsException e) {
+            DSpace dspace = new DSpace();
+            ConfigurationService configurationService = dspace.getConfigurationService();
+            String[] authMethods = configurationService.getArrayProperty("plugin.sequence.org.dspace.authenticate.AuthenticationMethod", new String[0]);
+            for(String method: authMethods) {
+                if (method.equals("org.dspace.authenticate.ShibAuthentication")) {
```

Comment:
I would prefer to not have a hard-coded reference to Shibboleth. This is important if, for example, we want to support CAS-based authentication in the future. CAS authentication can also redirect the user to an external login page similar to Shibboleth. And it would be nice that we can just "plugin" a DSpace CAS `AuthenticationMethod` implementation without the need to modify the REST API classes. Therefore I would change the code as follows:

1. Ask the login URL from each enabled authentication method:
```
AuthenticationMethod authenticationMethod = authenticationMethodIterator.next();
String loginPageURL = authenticationMethod.loginPageURL(context, req, res);
if (StringUtils.isNotBlank(loginPageURL)) {
    res.addHeader("Location", loginPageURL);
}
```

2. In DSpace API, adjust `org.dspace.authenticate.ShibAuthentication#loginPageURL` to use `req.getHeader("Referer")` as return url.

3. In DSpace API, adjust `org.dspace.authenticate.LDAPAuthentication` and `org.dspace.authenticate.PasswordAuthentication` to return `null` in their `loginPageURL()` method implementation. In DSpace 7 REST, these authentication methods do not really need a page as they have to pass the `user` and `password` parameters in the login request.